.github/workflows/main.yml

```diff
@@ -0,0 +1,70 @@
+name: Mac OS build
+
+on:
+  push
+
+jobs:
+  build:
+
+    runs-on: macos-latest
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Install dependencies
+      run: brew install gtk+3 gtkmm3 gtk-mac-integration adwaita-icon-theme libsigc++ little-cms2 libiptcdata fftw lensfun llvm expat pkgconfig libomp shared-mime-info
+    - name: patch libiconv
+      run: |
+        mkdir libiconv && cd libiconv
+        wget https://ftp.gnu.org/pub/gnu/libiconv/libiconv-1.16.tar.gz
+        tar xf libiconv-1.16.tar.gz
+        cd libiconv-1.16
+        patch -p1 < "${GITHUB_WORKSPACE}/tools/osx/libiconv_1.16_rt.patch"
+        mkdir build && cd build
+        destDir="$(pwd)"
+        ../configure --prefix=/opt/local --disable-static 'CFLAGS=-arch x86_64 -mmacosx-version-min=10.9' 'LDFLAGS=-arch x86_64 -mmacosx-version-min=10.9' CXXFLAGS="-arch x86_64 -mmacosx-version-min=10.9"
+        make --jobs 2
+        make DESTDIR="${destDir}" install
+        sudo mv opt/local /usr/local/opt/libiconv
+    - name: print /usr/local/lib contents
+      run: ls -al /usr/local/lib
+    - name: cmake
+      env:
+        CMAKE_CXX_STANDARD: 11
+        # clutching at the straw https://stackoverflow.com/questions/17912092/gtk-installed-with-brew-is-not-found-when-i-try-to-compile-other-applications with /opt/X11
+        PKG_CONFIG_PATH: /usr/local/opt/libffi/lib/pkgconfig:/usr/local/opt/expat/lib/pkgconfig:/opt/X11/lib/pkgconfig
+        RAW_THERAPEE_MAJOR: '5'
+        RAW_THERAPEE_MINOR: '7'
+        C_FLAGS: -Xpreprocessor -fopenmp /usr/local/lib/libomp.dylib -I/usr/local/include -I/usr/local/opt/gdk-pixbuf/include -I/usr/local/opt/libiconv/include -I/usr/local/opt/libxml2/include -I/usr/local/opt/expat/include -I/usr/local/opt/llvm/include
+      run: |
+        REF=${GITHUB_REF##*/}
+        mkdir build && cd build
+        cmake \
+          -DCMAKE_BUILD_TYPE="release" \
+          -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON \
+          -DCMAKE_EXE_LINKER_FLAGS="-L/usr/local/lib -L/usr/local/opt/llvm/lib -Wl,-rpath,/usr/local/opt/llvm/lib -L/usr/local/opt/gdk-pixbuf/lib -L/usr/local/opt/libiconv/lib -L/usr/local/opt/libffi/lib -L/usr/local/opt/libffi/lib -L/usr/local/opt/libxml2/lib -L/usr/local/opt/expat/lib" \
+          -DCACHE_NAME_SUFFIX="${RAW_THERAPEE_MAJOR}.${RAW_THERAPEE_MINOR}-${REF}" \
+          -DPROC_TARGET_NUMBER="2" \
+          -DPROC_LABEL="generic processor" \
+          -DWITH_LTO="OFF" \
+          -DLENSFUNDBDIR="./share/lensfun" \
+          -DOpenMP_C_FLAGS=-fopenmp=libomp \
+          -DOpenMP_CXX_FLAGS=-fopenmp=libomp \
+          -DOpenMP_C_LIB_NAMES="libomp" \
+          -DOpenMP_CXX_LIB_NAMES="libomp" \
+          -DOpenMP_libomp_LIBRARY="/usr/local/lib/libomp.dylib" \
+          -DOpenMP_C_FLAGS="${C_FLAGS}" \
+          -DOpenMP_CXX_FLAGS="${C_FLAGS}" \
+          -DCMAKE_AR="/usr/local/opt/llvm/bin/llvm-ar" \
+          -DCMAKE_RANLIB="/usr/local/opt/llvm/bin/llvm-ranlib" \
+          ..
+        make --jobs 2
```

Comment:
@Krijger can `jobs` not be left on automatic?

Reply:
I don't know. I saw it specified in every example I could find, so decided to include it here as well and specify the number of cores I know the build agent has. I guess we can simply remove it and see if the build does not become slower :)

Reply:
afaik `make -j` uses all available cores

Reply:
Nope - testing shows that removing `--jobs 2` makes the make step approximately twice as slow

Reply:
Yes, without `--jobs 2` only one core will be used.
I just tested with `-j` on my 8 core. In fact it starts much more than 8 jobs then...

Reply:
Tested the `--jobs` option to be as fast as `--jobs 2`. Keeping the first, since this will automatically improve the build if the agents grow in CPUs in the future.